realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -230,8 +227,57 @@ static void notifyUserLoggedOut(SyncUser user) {
         }
     }
 
+    // This is called from SyncManager.cpp from the worker thread the Sync Client is running on
+    // Right now Core doesn't send these errors to the proper session, so instead we need to notify all sessions
```

Comment:
Yes, this was fixed some time ago. See https://github.com/realm/realm-sync/blob/cfa258df3203cac5b9395769ab01764a730bfeb6/CHANGELOG.md#breaking-changes-18.